### ISSUE TITLE
EAPE values are huge

### ISSUE THREAD

**Issue by salvajoe** [salvajoe/thesis#9]
I found that my EAPE values are huge and not comparable with those reported by [Roullet et al. (2014)](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1002/2013GL059004) and [Luecke et al. (2017)](https://agupubs.onlinelibrary.wiley.com/doi/abs/10.1002/2017JC013136).

The only reason I found for it is density values obtained from the [WOA](https://www.ncei.noaa.gov/access/world-ocean-atlas-2018/bin/woa18.pl) that I use as the reference state.

But maybe there is something else I don't see. The notebook with the code is in \trials\WOA_density

**Comment by tkarna** [comment]
Yes it looks like your EAPE is several orders of magnitude larger. How did you define rho_0?

Looking at Luecke et al. (2017), it seems that they just use a constant value for rho_0. It appears in the linearised equation of state (eq. 2). Sadly I could not find which value they are using. Have you tried that kind of approach? You could just try with rho_0=1000 kg/m3 (or rho of pure sea water).

**Comment by salvajoe** [comment]
I defined rho_0 as the mean between two coherent isopycnal surfaces: rho_0 = (rho1+rho2)/2
This is how Petteri suggested last time we talked about the calculation of the EAPE. I also tried using the constant value, but it doesn't give any better result, since the same rho_0 value is used in N^2. 
The thing is that density values from the WOA that I use as the reference state are increasing with depth faster than the values derived from T,S measured by MMP. I think this might be a reason. So, I used the time-mean of density derived from the MMP measurements as a reference state. In this case, the values deeper than the halocline are still huge.

**Comment by tkarna** [comment]
I think one issue in your script is that `rhoreference` is around 27 kg/m3 when it should be 1027 kg/m3. It's the absolute density, not deviation (like rho').

**Comment by tkarna** [comment]
I've pushed an update to the trial branch. Basically N2 is too low in the interior ocean; as N2 appears in the denominator, you get very large values of EAPE. You get a much more consistent N2 if you compute it from the density instead of potential density anomaly sigma0 (in literature they use density). I also used a rho_0 constant in space and time (I do not see a reason why it should vary). The results are now closer to literature values.
![polina_new_eape](https://user-images.githubusercontent.com/7586775/128079184-7680e690-17a9-4b41-863a-1904a8019460.png)

**Comment by tkarna** [comment]
Hmm reading more carefully Luecke et al does define rho as potential density. But they do not explain in sufficient detail how they compute EAPE. In any case, you have N2 ~1e-7 in the deeper layers which leads to large EAPE values.

**Comment by tkarna** [comment]
I think the drawback of the EAPE equation you use is indeed that EAPE grows to infinity as N2 goes to zero (if rho' remains bounded). Your EAPE plot shows largest EAPE in the deep layers where N2 is small. Roullet et al fig4 shows that EAPE is close to zero in the deeper layers and maximum is in the stratified layer. This seems more consistent with the theory. Maybe it would be better to use a method like Roullet et al., as they also compute EAPE for vertical profiles.

In the introduction, Kang and Fringer (2010) also mention another EAPE_3 formula:
EAPE = rho_0 N2 zeta^2 / 2
As mentioned in Luecke, this should be scaled by 1/rho_0, so you'd get
EAPE = N2 zeta^2 / 2
This form has the nice property that EAPE is 0 for N2=0. But you'd have to calculate the vertical displacement zeta (wrt the reference density profile).

**Comment by salvajoe** [comment]
Thanks, Tuomas! 
I also re-did the plot yesterday using the density values (not sigma) and got a better result. I tried to use the constant value rho_0 = 1025 and rho_0 = (rho1+rho2)/2: they both work since there is rho_0 in the N^2 equation and in the EAPE. 
In addition, I calculated rho_anomaly as the departure from the time mean measured by MMP instead of using the density values from the World Ocean Atlas. Again, Luecke uses a time mean of density and don't explicitly explain what time mean. I initially thought it should be a climatological mean and therefore used the WOA values. But as I mentioned already, the WOA density increases too fast reaching 1031 kg/m^3 at depth 700 while MMP density is max. 28.04 kg/m^3. So, using the time-mean measured by MMP gives the following picture. I also converted m^2/s^2 to cm^2/s^2 to easier compare the values with those in Luecke. 
![EAPE_timeseries](https://user-images.githubusercontent.com/10695922/128134994-0014ea2b-cffe-43f4-92d4-f86d15add4dd.png)

**Comment by tkarna** [comment]
Nice, that highlights the eddy time period nicely. Yes, I think it makes sense to use the time-averaged density profile of the observations as the reference profile. EAPE does still depend on rho_0, no?  Substituting N2 into the equation still leaves one rho_0 in the denominator. It probably does not matter much which rho_0 you use (as long as it's ~1025 and not ~25 kg/m3).

**Comment by salvajoe** [comment]
Yes, one rho_0 remains in the denominator, but the most important is to use the absolute density value ~10^3 kg/m3 instead of sigma_0 ~10^1 kg/m3. But constant or variable is already not much important.
